### ISSUE TITLE
Getting app crash after updating to 2.18.2

### ISSUE THREAD

**Issue by dev-sharma-deltatre** [google/ExoPlayer#10828]
### ExoPlayer Version

2.18.2

### Devices that reproduce the issue

Android Tv and Fire TV

### Reproducible in the demo app?

Yes

### Reproduction steps

Play any video

### Expected result

Vidoe should play

### Actual result

App is crashing

### Media

   --------- beginning of crash
2022-12-02 17:04:52.135 E: FATAL EXCEPTION: pool-31-thread-2
    Process: ca.tsn.mobile.android.stag, PID: 22224
    com.google.ads.interactivemedia.v3.internal.ayb: java.lang.IncompatibleClassChangeError: The method 'java.lang.Class java.lang.Object.getClass()' was expected to be of type interface but instead was found to be of type virtual (declaration of 'com.google.ads.interactivemedia.v3.internal.bqd' appears in /data/app/ca.tsn.mobile.android.stag-1/base.apk:classes17.dex)
        at com.google.ads.interactivemedia.v3.internal.ayl.b(IMASDK:6)
        at com.google.ads.interactivemedia.v3.internal.ayu.e(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.te.run(IMASDK:5)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
        at java.lang.Thread.run(Thread.java:761)
     Caused by: java.lang.IncompatibleClassChangeError: The method 'java.lang.Class java.lang.Object.getClass()' was expected to be of type interface but instead was found to be of type virtual (declaration of 'com.google.ads.interactivemedia.v3.internal.bqd' appears in /data/app/ca.tsn.mobile.android.stag-1/base.apk:classes17.dex)
        at com.google.ads.interactivemedia.v3.internal.bqd.<clinit>(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.bqf.<clinit>(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.bqf.f(IMASDK)
        at com.google.ads.interactivemedia.v3.internal.bqj.a(IMASDK:7)
        at com.google.ads.interactivemedia.v3.internal.bqx.b(IMASDK:3)
        at com.google.ads.interactivemedia.v3.internal.bqx.c(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.bpg.aW(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.bpg.aV(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.aij.m(IMASDK:7)
        at com.google.ads.interactivemedia.v3.internal.aij.f(IMASDK:2)
        at com.google.ads.interactivemedia.v3.internal.aif.call(IMASDK)
        at com.google.ads.interactivemedia.v3.internal.ayw.a(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.ayp.run(IMASDK:4)
        at com.google.ads.interactivemedia.v3.internal.ayx.run(IMASDK:1)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607) 
        at java.lang.Thread.run(Thread.java:761) 

### Bug Report

- [x] You will email the zip file produced by `adb bugreport` to dev.exoplayer@gmail.com after filing this issue.

**Comment by rohitjoins** [comment]
Hi. I am able to play videos on our demo app. Seeing the stack trace it is hard to tell if this exception is coming from Exoplayer. When is this exactly happening(first time classes are loaded is the guess)? Do you also define your own dependency to IMA? Can you provide a minimal app on which this happens?

**Comment by dev-sharma-deltatre** [comment]
@rohitjoins No, I am only using ima- extension by exoplyer.  Yes, it is happening when first-time classes are loading.  I cannot provide the app as it is already in market. But from the logs it looks like something from com.google.ads.interactivemedia.v3.internal.ayb this package which expects class as an interface. Can you help me understand this issue more.

**Comment by dev-sharma-deltatre** [comment]
The logs that I have pasted previously were from FireTv. attaching the logs from Android Tv as well.
2022-12-05 14:06:02.237 E: FATAL EXCEPTION: pool-35-thread-1
    Process: ca.tsn.mobile.android.stag, PID: 10342
    com.google.ads.interactivemedia.v3.internal.ayb: java.lang.ExceptionInInitializerError
        at com.google.ads.interactivemedia.v3.internal.ayl.b(IMASDK:6)
        at com.google.ads.interactivemedia.v3.internal.ayu.e(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.te.run(IMASDK:5)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:923)
     Caused by: java.lang.ExceptionInInitializerError
        at com.google.ads.interactivemedia.v3.internal.bqf.<clinit>(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.bqf.f(Unknown Source:0)
        at com.google.ads.interactivemedia.v3.internal.bqj.a(IMASDK:7)
        at com.google.ads.interactivemedia.v3.internal.bqx.b(IMASDK:3)
        at com.google.ads.interactivemedia.v3.internal.bqx.c(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.bpg.aW(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.bpg.aV(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.aij.m(IMASDK:7)
        at com.google.ads.interactivemedia.v3.internal.aij.f(IMASDK:2)
        at com.google.ads.interactivemedia.v3.internal.aif.call(Unknown Source:4)
        at com.google.ads.interactivemedia.v3.internal.ayw.a(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.ayp.run(IMASDK:4)
        at com.google.ads.interactivemedia.v3.internal.ayx.run(IMASDK:1)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:923) 
     Caused by: java.lang.NullPointerException: Attempt to read from field 'java.util.Collection java.util.Collections$UnmodifiableCollection.c' on a null object reference
        at java.util.Collections$UnmodifiableCollection.toArray(Collections.java:1081)
        at com.google.ads.interactivemedia.v3.internal.bqd.<clinit>(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.bqf.<clinit>(IMASDK:1) 
        at com.google.ads.interactivemedia.v3.internal.bqf.f(Unknown Source:0) 
        at com.google.ads.interactivemedia.v3.internal.bqj.a(IMASDK:7) 
        at com.google.ads.interactivemedia.v3.internal.bqx.b(IMASDK:3) 
        at com.google.ads.interactivemedia.v3.internal.bqx.c(IMASDK:1) 
        at com.google.ads.interactivemedia.v3.internal.bpg.aW(IMASDK:1) 
        at com.google.ads.interactivemedia.v3.internal.bpg.aV(IMASDK:1) 
        at com.google.ads.interactivemedia.v3.internal.aij.m(IMASDK:7) 
        at com.google.ads.interactivemedia.v3.internal.aij.f(IMASDK:2) 
        at com.google.ads.interactivemedia.v3.internal.aif.call(Unknown Source:4) 
        at com.google.ads.interactivemedia.v3.internal.ayw.a(IMASDK:1) 
        at com.google.ads.interactivemedia.v3.internal.ayp.run(IMASDK:4) 
        at com.google.ads.interactivemedia.v3.internal.ayx.run(IMASDK:1) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:923)

**Comment by rohitjoins** [comment]
@marcbaechinger Can you please look into this? We're unsure of how much we can do based only on the logs.

**Comment by dev-sharma-deltatre** [comment]
@marcbaechinger Any update?

**Comment by palvaibhav89** [comment]
I am also facing the same issue in my app. 

This is happening on 2.18.2

2023-01-16 12:53:58.475 19080-19718/xxxxx E/AndroidRuntime: FATAL EXCEPTION: pool-67-thread-1
    Process: , PID: 19080
    com.google.ads.interactivemedia.v3.internal.ayb: java.lang.ExceptionInInitializerError
        at com.google.ads.interactivemedia.v3.internal.ayl.b(IMASDK:6)
        at com.google.ads.interactivemedia.v3.internal.ayu.e(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.te.run(IMASDK:5)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:923)
     Caused by: java.lang.ExceptionInInitializerError
        at com.google.ads.interactivemedia.v3.internal.bqf.<clinit>(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.bqf.f(Unknown Source:0)
        at com.google.ads.interactivemedia.v3.internal.bqj.a(IMASDK:7)
        at com.google.ads.interactivemedia.v3.internal.bqx.b(IMASDK:3)
        at com.google.ads.interactivemedia.v3.internal.bqx.c(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.bpg.aW(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.bpg.aV(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.aij.m(IMASDK:7)
        at com.google.ads.interactivemedia.v3.internal.aij.f(IMASDK:2)
        at com.google.ads.interactivemedia.v3.internal.aif.call(Unknown Source:4)
        at com.google.ads.interactivemedia.v3.internal.ayw.a(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.ayp.run(IMASDK:4)
        at com.google.ads.interactivemedia.v3.internal.ayx.run(IMASDK:1)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:923)
     Caused by: java.lang.NullPointerException: Attempt to read from field 'java.util.Collection java.util.Collections$UnmodifiableCollection.c' on a null object reference
        at java.util.Collections$UnmodifiableCollection.toArray(Collections.java:1081)
        at com.google.ads.interactivemedia.v3.internal.bqd.<clinit>(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.bqf.<clinit>(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.bqf.f(Unknown Source:0)
        at com.google.ads.interactivemedia.v3.internal.bqj.a(IMASDK:7)
        at com.google.ads.interactivemedia.v3.internal.bqx.b(IMASDK:3)
        at com.google.ads.interactivemedia.v3.internal.bqx.c(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.bpg.aW(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.bpg.aV(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.aij.m(IMASDK:7)
        at com.google.ads.interactivemedia.v3.internal.aij.f(IMASDK:2)
        at com.google.ads.interactivemedia.v3.internal.aif.call(Unknown Source:4)
        at com.google.ads.interactivemedia.v3.internal.ayw.a(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.ayp.run(IMASDK:4)
        at com.google.ads.interactivemedia.v3.internal.ayx.run(IMASDK:1)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:923)

**Comment by ArsalImam** [comment]
+1, getting the same crash on Exoplayer v2.18.2. Working fine on older versions (< 2.18.2) of exoplayer.

```
2023-01-16 12:53:58.475 19080-19718/noice.app.qa E/AndroidRuntime: FATAL EXCEPTION: pool-67-thread-1
    Process: [noice.app.qa](http://noice.app.qa/), PID: 19080
    com.google.ads.interactivemedia.v3.internal.ayb: java.lang.ExceptionInInitializerError
        at com.google.ads.interactivemedia.v3.internal.ayl.b(IMASDK:6)
        at com.google.ads.interactivemedia.v3.internal.ayu.e(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.te.run(IMASDK:5)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:923)
     Caused by: java.lang.ExceptionInInitializerError
        at com.google.ads.interactivemedia.v3.internal.bqf.<clinit>(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.bqf.f(Unknown Source:0)
        at com.google.ads.interactivemedia.v3.internal.bqj.a(IMASDK:7)
        at com.google.ads.interactivemedia.v3.internal.bqx.b(IMASDK:3)
        at com.google.ads.interactivemedia.v3.internal.bqx.c(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.bpg.aW(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.bpg.aV(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.aij.m(IMASDK:7)
        at com.google.ads.interactivemedia.v3.internal.aij.f(IMASDK:2)
        at com.google.ads.interactivemedia.v3.internal.aif.call(Unknown Source:4)
        at com.google.ads.interactivemedia.v3.internal.ayw.a(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.ayp.run(IMASDK:4)
        at com.google.ads.interactivemedia.v3.internal.ayx.run(IMASDK:1)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:923)
     Caused by: java.lang.NullPointerException: Attempt to read from field 'java.util.Collection java.util.Collections$UnmodifiableCollection.c' on a null object reference
        at java.util.Collections$UnmodifiableCollection.toArray(Collections.java:1081)
        at com.google.ads.interactivemedia.v3.internal.bqd.<clinit>(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.bqf.<clinit>(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.bqf.f(Unknown Source:0)
        at com.google.ads.interactivemedia.v3.internal.bqj.a(IMASDK:7)
        at com.google.ads.interactivemedia.v3.internal.bqx.b(IMASDK:3)
        at com.google.ads.interactivemedia.v3.internal.bqx.c(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.bpg.aW(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.bpg.aV(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.aij.m(IMASDK:7)
        at com.google.ads.interactivemedia.v3.internal.aij.f(IMASDK:2)
        at com.google.ads.interactivemedia.v3.internal.aif.call(Unknown Source:4)
        at com.google.ads.interactivemedia.v3.internal.ayw.a(IMASDK:1)
        at com.google.ads.interactivemedia.v3.internal.ayp.run(IMASDK:4)
        at com.google.ads.interactivemedia.v3.internal.ayx.run(IMASDK:1)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:923)
```

**Comment by marcbaechinger** [comment]
This is a bug in the IMA SDK version 2.28.1 that the [IMA extension is depending on](https://github.com/androidx/media/blob/release/libraries/exoplayer_ima/build.gradle). We bumped the version in [the main branch already](https://github.com/androidx/media/blob/main/libraries/exoplayer_ima/build.gradle) which should removes the issue. This will be included in the next release.

EDIT: Sorry, the links are pointing to the androix/media repo. The fix is in both repos in the dev branch (`dev-v2` for `google/exoplayer` and `main` for `androidx/media`).

**Comment by dev-sharma-deltatre** [comment]
@marcbaechinger  Could you please provide any tentative date for releasing these fixes?

**Comment by marcbaechinger** [comment]
We are currently in the process of releasing. I guess it will be start of February or similar until the release is published.

**Comment by dev-sharma-deltatre** [comment]
@marcbaechinger Hello, It is February, I just wanted to remind you about the release for this fix as the IMA SDK version used in 2.18.2 is going to be deprecated soon. Eagerly waiting for the release.
